pkgs/development/libraries/gstreamer/gstreamermm/default.nix

```
@@ -1,26 +1,62 @@
-{ lib, stdenv, fetchurl, pkg-config, file, glibmm, gst_all_1, gnome }:
+{ lib
+, stdenv
+, fetchFromGitLab
+, autoreconfHook
+, perl
+, mm-common
+, pkg-config
+, file
+, glibmm
+, gst_all_1
+, nix-update-script
+}:
+
 stdenv.mkDerivation rec {
   pname = "gstreamermm";
-  version = "1.10.0";
+  version = "unstable-2022-05-03";
 
-  src = fetchurl {
-    url = "mirror://gnome/sources/${pname}/${lib.versions.majorMinor version}/${pname}-${version}.tar.xz";
-    sha256 = "0q4dx9sncqbwgpzma0zvj6zssc279yl80pn8irb95qypyyggwn5y";
+  src = fetchFromGitLab {
+    # https://gitlab.gnome.org/GNOME/gstreamermm
+    domain = "gitlab.gnome.org";
+    owner = "GNOME";
+    repo = "gstreamermm";
+    # https://gitlab.gnome.org/GNOME/gstreamermm/-/merge_requests/5
+    rev = "89b2ece5cab32b5094686bb96426a004029a9449";
+    sha256 = "sha256-b6719eTiBJ21eNA/tgul6GuAkKF0gmsbbG2x9NAtW3s=";
   };
 
   outputs = [ "out" "dev" ];
 
-  nativeBuildInputs = [ pkg-config file ];
+  nativeBuildInputs = [
+    autoreconfHook
+    mm-common
+    pkg-config
+    file
+  ];
 
-  propagatedBuildInputs = [ glibmm gst_all_1.gst-plugins-base ];
+  propagatedBuildInputs = [
+    glibmm
+    gst_all_1.gst-plugins-base
+    gst_all_1.gst-plugins-good
+  ];
 
   enableParallelBuilding = true;
 
+  GMMPROC_DIR = "${glibmm}/lib/glibmm-2.4/proc";
+  PERL = (perl.withPackages (p: [ p.XMLParser ])) + "/bin/perl";
+
+  # must run before autoreconfPhase
+  postPatch = ''
+    autoupdate
+    sed -i '/^autoreconf /d' autogen.sh
+    NOCONFIGURE=1 ./autogen.sh
+  '';
+
+  configureFlags = [ "--enable-maintainer-mode" ]; # from autogen.sh
```

Maybe something like:

```suggestion
  configureFlags = [
    # Enable generating of binding code (pre-generated C++ files not available in Git).
    "--enable-maintainer-mode"
  ];
```